drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c

```diff
@@ -0,0 +1,488 @@
+
+	config->flexio_lcd_dev->rxShifterStartIndex = config->flexio_lcd_dev->txShifterStartIndex;
+	config->flexio_lcd_dev->rxShifterEndIndex = config->flexio_lcd_dev->txShifterEndIndex;
+
+	config->flexio_lcd_dev->timerIndex = config->child->res.timer_index[0];
+
+	if (config->flexio_lcd_dev->txShifterEndIndex !=
+	    config->flexio_lcd_dev->txShifterStartIndex + shifter_end) {
+		LOG_ERR("Shifters should be continuous");
+		return -ENODEV;
+	}
+	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+	if (err) {
+		return err;
+	}
+
+	/* Pass the FlexIO LCD device as parameter to the function
+	 * callbacks for setting GPIO signals.
+	 */
+	config->flexio_lcd_dev->userData = (void *)dev;
+
+	k_sem_init(&lcdif_data->transfer_done, 0, 1);
+
+	LOG_DBG("%s device init complete", dev->name);
+
+	return 0;
+}
+
+static struct mipi_dbi_driver_api mipi_dbi_lcdif_driver_api = {
+	.reset = mipi_dbi_flexio_lcdif_reset,
+	.command_write = mipi_dbi_flexio_lcdif_command_write,
+	.write_display = mipi_dbi_flexio_ldcif_write_display,
+};
+
+/* The width is read from device tree by the CMake file and passed in as a
+ * compile time define
+ */
+#if (8UL == FLEXIO_MCULCD_DATA_BUS_WIDTH)
+#define MCUX_FLEXIO_SOURCE_DATA_SIZE	.source_data_size = 1,
+#define MCUX_FLEXIO_DATA_BUS_WIDTH	.data_bus_width = 8,
+#else
+#define MCUX_FLEXIO_SOURCE_DATA_SIZE	.source_data_size = 2,
+#define MCUX_FLEXIO_DATA_BUS_WIDTH	.data_bus_width = 16,
+#endif
+
+#define MCUX_FLEXIO_LCDIF_DEVICE_INIT(n)						\
+											\
+	PINCTRL_DT_INST_DEFINE(n);							\
+											\
+	static FLEXIO_MCULCD_Type flexio_mculcd_##n = {					\
+		.flexioBase = (FLEXIO_Type *)DT_REG_ADDR(DT_INST_PARENT(n)),		\
+		.dataPinStartIndex = DT_INST_PROP(n, data_pin_start),			\
+		.ENWRPinIndex = DT_INST_PROP(n, enwr_pin),				\
+		.RDPinIndex = DT_INST_PROP_OR(n, rd_pin, 0),				\
+		.setCSPin = flexio_lcdif_set_cs,					\
+		.setRSPin = flexio_lcdif_set_rs,					\
+		.setRDWRPin = flexio_lcdif_set_rd_wr,					\
+	};										\
+											\
+	static uint8_t mcux_flexio_lcdif_shifters_##n[DT_INST_PROP(n, shifters_count)];	\
+	static uint8_t mcux_flexio_lcdif_timers_##n[DT_INST_PROP(n, timers_count)];	\
+											\
+	static const struct nxp_flexio_child lcdif_child_##n = {			\
+		.isr = flexio_lcdif_isr,						\
+		.user_data = (void *)DEVICE_DT_INST_GET(n),				\
+		.res = {								\
+			.shifter_index = mcux_flexio_lcdif_shifters_##n,		\
+			.shifter_count = ARRAY_SIZE(mcux_flexio_lcdif_shifters_##n),	\
+			.timer_index = mcux_flexio_lcdif_timers_##n,			\
+			.timer_count = ARRAY_SIZE(mcux_flexio_lcdif_timers_##n),	\
+		}									\
+	};										\
+											\
+	struct mcux_flexio_lcdif_config mcux_flexio_lcdif_config_##n = {		\
+		.flexio_lcd_dev = &flexio_mculcd_##n,					\
+		.flexio_dev = DEVICE_DT_GET(DT_INST_PARENT(n)),				\
+		.pincfg = PINCTRL_DT_INST_DEV_CONFIG_GET(n),				\
+		.child = &lcdif_child_##n,						\
+		.baudrate_bps = DT_INST_PROP(n, baudrate_bps),				\
```

Comment:
See comment above- I think we should calculate the baudrate within the `configure` function above, based on the display's requested frequency.

Reply:
I have deleted this property